### PR TITLE
use id and idlist as attribute types

### PR DESCRIPTION
**Pull request by xi**

### CODE REVIEW

**Comment by aabenoja** [comment]
Pretty sure this will break [eslint-plugin-jsx-a11y](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/3fb2583b2573199bb49e578e0d2415bf819c272b/src/rules/aria-proptypes.js#L32-L53).

**Comment by jessebeach** [comment]
@aabenoja interesting change. I like the direction. I'm trying to move us to an opaque type `DomID` internally so that we're not simply accepting string where we want DOM IDs. Let me think about this some more. I want to investigate your hypothesis that this will break eslint-plugin-jsx-a11y. We should be able to make it work.

**Comment by jessebeach** [comment]
@xi apologies for the long gap in response. I appreciate the intent of this change. It needs a little work. Flow types must eventually unpack to known types for scalars (number, boolean, string, etc) and declare types for classes and object shapes. In order to introduce a type `id` or `idlist`, we would need to alias it to the opaque type `string`. You can do that with Flow opaque types:

https://flow.org/en/docs/types/opaque-types/

Punting back to you @xi for updates.

@aabenoja and yes, we would need to update @eslint-plugin-jsx-a11y to understand that `id` and `idlist` are aliased to `string`.

**Comment by xi** [comment]
I am not sure uf I understand correctly. As far as I understand, my change does not mean that `ARIAPropertyDefinition.type` needs to of type `string` or `id` or …. Instead it means that it `ARIAPropertyDefinition.type` needs to **be** one of the strings `string` or `id` or …. I am not trying to introduce a new flow type here.

File: flow/aria.js [hunk 1]
```diff
@@ -269,6 +269,8 @@ type ARIAPropertyMap = {
 
 type ARIAPropertyDefinition = {
   type: 'string'
+  | 'id',
+  | 'idlist',
```
Comment (aabenoja) [inline flow/aria.js]: On an interesting note, the [build report](https://travis-ci.org/A11yance/aria-query/jobs/346199294#L848-L851) is passing despite these commas causing parsing errors for the flow check. @jessebeach

**Comment by jessebeach** [comment]
> I am not sure if I understand correctly. As far as I understand, my change does not mean that ARIAPropertyDefinition.type needs to be of type string or id or …. Instead it means that it ARIAPropertyDefinition.type needs to be one of the strings 'string' or 'id' or …. I am not trying to introduce a new flow type here.

Indeed, yes, I'm wrong here. Thank you for persisting with the explanation. Thank you for the PR!

**Pull request merged**